compiler/src/dmd/enumsem.d

```diff
@@ -0,0 +1,714 @@
+/**
+ * Does the semantic passes on enums.
+ *
+ * Copyright:   Copyright (C) 1999-2024 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 https://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 https://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/enumsem.d, _enumsem.d)
+ * Documentation:  https://dlang.org/phobos/dmd_enumsem.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/enumsem.d
+ */
+
+module dmd.enumsem;
+
+import core.stdc.stdio;
+import core.stdc.string;
+
+import dmd.aggregate;
+import dmd.aliasthis;
+import dmd.arraytypes;
+import dmd.astcodegen;
+import dmd.astenums;
+import dmd.attrib;
+import dmd.blockexit;
+import dmd.clone;
+import dmd.cond;
+import dmd.compiler;
+import dmd.dcast;
+import dmd.dclass;
+import dmd.declaration;
+import dmd.denum;
+import dmd.dimport;
+import dmd.dinterpret;
+import dmd.dmangle;
+import dmd.dmodule;
+import dmd.dscope;
+import dmd.dstruct;
+import dmd.dsymbol;
+import dmd.dsymbolsem;
+import dmd.dtemplate;
+import dmd.dversion;
+import dmd.errors;
+import dmd.escape;
+import dmd.expression;
+import dmd.expressionsem;
+import dmd.func;
+import dmd.funcsem;
+import dmd.globals;
+import dmd.id;
+import dmd.identifier;
+import dmd.importc;
+import dmd.init;
+import dmd.initsem;
+import dmd.intrange;
+import dmd.hdrgen;
+import dmd.location;
+import dmd.mtype;
+import dmd.mustuse;
+import dmd.nogc;
+import dmd.nspace;
+import dmd.objc;
+import dmd.opover;
+import dmd.optimize;
+import dmd.parse;
+import dmd.root.array;
+import dmd.root.filename;
+import dmd.common.outbuffer;
+import dmd.root.rmem;
+import dmd.rootobject;
+import dmd.root.utf;
+import dmd.semantic2;
+import dmd.semantic3;
+import dmd.sideeffect;
+import dmd.statementsem;
+import dmd.staticassert;
+import dmd.tokens;
+import dmd.utils;
+import dmd.statement;
+import dmd.target;
+import dmd.templateparamsem;
+import dmd.typesem;
+import dmd.visitor;
+
+
+/*********************************
+ * Perform semantic analysis on enum declaration `em`
+ */
+void enumSemantic(Scope* sc, EnumDeclaration ed)
+{
+    //printf("EnumDeclaration::semantic(sd = %p, '%s') %s\n", sc.scopesym, sc.scopesym.toChars(), ed.toChars());
+    //printf("EnumDeclaration::semantic() %p %s\n", ed, ed.toChars());
+    if (ed.semanticRun >= PASS.semanticdone)
+	return; // semantic() already completed
+    if (ed.semanticRun == PASS.semantic)
+    {
+	assert(ed.memtype);
```

Comment:
Inconsistent indentation, using tabs instead of spaces

Reply:
fixed